pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -1045,28 +1045,14 @@ func (p *PriorityQueue) AssignedPodAdded(logger klog.Logger, pod *v1.Pod) {
 	p.lock.Unlock()
 }
 
-// isPodResourcesResizedDown returns true if a pod CPU and/or memory resize request has been
-// admitted by kubelet, is 'InProgress', and results in a net sizing down of updated resources.
-// It returns false if either CPU or memory resource is net resized up, or if no resize is in progress.
-func isPodResourcesResizedDown(pod *v1.Pod) bool {
-	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
-		// TODO(vinaykul,wangchen615,InPlacePodVerticalScaling): Fix this to determine when a
-		// pod is truly resized down (might need oldPod if we cannot determine from Status alone)
-		if pod.Status.Resize == v1.PodResizeStatusInProgress {
-			return true
-		}
-	}
-	return false
-}
-
 // AssignedPodUpdated is called when a bound pod is updated. Change of labels
 // may make pending pods with matching affinity terms schedulable.
-func (p *PriorityQueue) AssignedPodUpdated(logger klog.Logger, oldPod, newPod *v1.Pod) {
+func (p *PriorityQueue) AssignedPodUpdated(logger klog.Logger, oldPod, newPod *v1.Pod, event framework.ClusterEvent) {
 	p.lock.Lock()
-	if isPodResourcesResizedDown(newPod) {
```

Comment:
Plz add back the todo here because what we really want here is **size down** if I understand correctly.

Reply:
Yes, but that's already handled by QHint of the resource fit plugin. So, nothing else to do, I think.

Reply:
Fair to me now because we didn't make the code worse, we didn't implement **resize down** logic before as well.

But in the long run, I think this is a matter of choice, we can always leave the judgement to the hint functions, e.g. replace below with `p.moveAllToActiveOrBackoffQueue` 

```
p.movePodsToActiveOrBackoffQueue(logger, p.getUnschedulablePodsWithMatchingAffinityTerm(logger, newPod), AssignedPodUpdate, oldPod, newPod)
```

**OR** we have another **simple** filter logic ahead of hint function like we will judge _whether resize downed here_ instead of _whether the request changed_ for the sake of a bit of performance, maybe huge in a large cluster.

Poor points.

Reply:
I agree with the first point; we should remove all pre-filtering stuff (including your example and preCheck) and always leave the judgement to QHint.

Reply:
what if QHints are disabled? Is there anything else filtering for size down specifically?

Reply:
> Is there anything else filtering for size down specifically?

In case QHint is disabled, No.

But, this topic is closely related to https://github.com/kubernetes/kubernetes/pull/122628#discussion_r1593835035
> @alculquicondor What do you think; UpdatePodRequest vs PodRequestScaledDown. Any particular concern for PodRequestScaledDown?

Maybe we can define `PodRequestScaledDown` instead of `UpdatePodRequest`. What do you think? @alculquicondor

Reply:
Yes, let's do that.

Reply:
Done.